include/RAJA/pattern/tensor/internal/TensorTileExec.hpp

```diff
@@ -152,6 +205,138 @@ namespace expt
 
     }
 
+
+    template<typename STORAGE, typename DIM_SEQ, typename IDX_SEQ>
+    struct StaticTensorTileExec;
+
+    /**
+     * Implement a dimension tiling loop
+     */
+
+    template<typename STORAGE, camp::idx_t DIM0, camp::idx_t ... DIM_REST, camp::idx_t IDX, camp::idx_t ... IDX_REST>
+    struct StaticTensorTileExec<STORAGE, camp::idx_seq<DIM0, DIM_REST...>,camp::idx_seq<IDX,IDX_REST...>>{
+
+          using DimList  = camp::idx_seq<DIM0, DIM_REST...>;
+          using DimTail  = camp::idx_seq<      DIM_REST...>;
+          using IdxList  = camp::idx_seq<IDX , IDX_REST...>;
+          using IdxTail  = camp::idx_seq<      IDX_REST...>;
+
+          using DownExec = TensorTileExec<STORAGE,camp::idx_seq<DIM_REST...>>;
+          using NextExec = StaticTensorTileExec<STORAGE,camp::idx_seq<DIM0,DIM_REST...>,camp::idx_seq<IDX_REST...>>;
+
+          static auto const step_size = STORAGE::s_dim_elem(DIM0);
+
+          template<
+              typename OTILE,
+              typename TTYPE,
+              typename BODY
+          >
+          RAJA_HOST_DEVICE
+          RAJA_INLINE
+          static
+          void
+          exec(
+              OTILE const &otile,
+              TTYPE const &tile,
+              BODY && body
+          ){
+    
+            auto constexpr orig_begin = OTILE::begin_type::value_at(DIM0);
+            auto constexpr orig_size =  OTILE:: size_type::value_at(DIM0);
+    
+            auto constexpr tile_begin = TTYPE::begin_type::value_at(DIM0);
+
+            using NextBegin = camp::integral_constant<typename TTYPE::index_type,tile_begin+STORAGE::s_dim_elem(DIM0)>;
+            using TailSize  = camp::integral_constant<typename TTYPE::index_type,(orig_begin+orig_size)-tile_begin>;
+
+            using NextTile  = typename expt::SetStaticTensorTileBegin<TTYPE,NextBegin,(size_t)DIM0>::Type;
+
+            using TailTile  = typename expt::SetStaticTensorTileSize <TTYPE,TailSize ,(size_t)DIM0>::Type;
+            using PartTile  = typename TailTile::Partial;
+
+    
+            static_assert( (tile_begin + STORAGE::s_dim_elem(DIM0) ) <= (orig_begin + orig_size+ STORAGE::s_dim_elem(DIM0) ), "OOB" );
+     
+            if( (tile_begin + STORAGE::s_dim_elem(DIM0) ) <= (orig_begin + orig_size) ){
+               DownExec::static_exec(otile, tile, body);
+               NextTile next_tile;
+               NextExec::exec(otile, next_tile, body);
+            } else if ( tile_begin < (orig_begin + orig_size ) ) {
+               PartTile part_tile;
+               DownExec::static_exec(otile,part_tile,body);
+            }
+    
+          }
```

Comment:
Is there an ordering constraint that requires that this be left-to-right top-down ordered? I'd have to sit and stare at this for a while to figure out how exactly to compose it, but if the final `DownExec::static_exec` is the part we care about, and it requires these instantiations but not this order, could we get away with a 2-dimensional pack expansion for this?

Something like:

```c++
// takes < DIM0, IdxList  > and does another pack expand on IDXs
(StaticTileExecHelper<DIMs, IdxList>::static_exec(otile, expt::SetStaticTensorTileSize <TTYPE,TailSize ,(size_t)DIMs>::Type, body), ...)
```

I'm using the C++17 comma fold for this, but we could do it with a null function or whatever too.  If it's valid to take the ordering out like this, it would tell the compiler very explicitly that we do not care about ordering and maybe free it up to use more registers.